models/demos/falcon7b_common/tt/falcon_attention.py

```diff
@@ -51,61 +50,38 @@ def __init__(
         # Different from paper, but it uses a different permutation in order to obtain the same calculation
         emb = torch.cat((freqs, freqs), dim=-1)
 
-        layer_name = f"{base_url}.{layer_num}.rotary_embedding"
+        layer_name = f"{base_url}.{layer_num}.rotary_embedding_maxseq{max_position_embeddings}"
         cos_str = f"{layer_name}.cos_cached"
         sin_str = f"{layer_name}.sin_cached"
 
-        overwrite_cos, overwrite_sin = False, False
-
-        for _ in range(2):
-            self.tt_cos_cached = get_weights_cached(
-                tt_devices,
-                model_config,
-                tt_cache_path,
-                cos_str,
-                weight_config_str="COS_CACHED_WEIGHTS",
-                weights_to_cache=emb.cos()[None, None, :, :],
-                overwrite=overwrite_cos,
-            )
-            overwrite_cos = (
-                tt2torch_tensor(self.tt_cos_cached[0]).shape[-2] != self.max_seq_len_cached
-            )  # Verify cached tensor has same max seq len
-            if not overwrite_cos:
-                break
-
-        for _ in range(2):
-            self.tt_sin_cached = get_weights_cached(
-                tt_devices,
-                model_config,
-                tt_cache_path,
-                sin_str,
-                weight_config_str="SIN_CACHED_WEIGHTS",
-                weights_to_cache=emb.sin()[None, None, :, :],
-                overwrite=overwrite_sin,
-            )
-            overwrite_sin = (
-                tt2torch_tensor(self.tt_sin_cached[0]).shape[-2] != self.max_seq_len_cached
-            )  # Verify cached tensor has same max seq len
-            if not overwrite_sin:
-                break
+        self.tt_cos_cached = get_weights_cached(
```

Comment:
I'd strongly recommend swapping arguments of the method where you first provide weights_to_cache and name for it. device_mesh, model_config should be later args.